foglamp/FogLAMP
# Fogl 261/262 - Rest api support for Schedules and Tasks - Read function only

With reference to [Rest API Architecture Doc](https://docs.google.com/docuement/d/1JJDP7g25SWerNVCxgff02qp9msHbqA9nt3RAFx8-Qng/edit#heading=h.al9srgd46ohj), Rest api support for read functions for Schedules and Tasks provided. Create/Update/Delete api will be provided when required support becomes available from Scheduler module.

To test this PR, please proceed as follows:
- Create DB and populate the tables per SQL provided. Also, please do incorporate scheduler_demo_data.sql. ~~You may be required to do some schedule_time adjustments in order to run the Scheduler.~~
- If required, modify DB connection string, per your local setting, in scheduler.py and scheduler_db_services.py.
- Generate Tasks data by invoking the Scheduler.
- Now you can check GET urls from the above Architecture doc for Schedules and Tasks e.g.
http://localhost:8082/foglamp/tasks
http://localhost:8082/foglamp/tasks?name=xxxx&state=yyyy
http://localhost:8082/foglamp/tasks?name=xxxx
http://localhost:8082/foglamp/tasks?state=yyyy
http://localhost:8082/foglamp/tasks/latest
http://localhost:8082/foglamp/tasks/latest?name=xxxx
http://localhost:8082/foglamp/task/<task_id>
http://localhost:8082/foglamp/schedules
http://localhost:8082/foglamp/schedule/<schedule_id>
